### ISSUE TITLE
Use `std::any`

### ISSUE THREAD

**Issue by ptheywood** [FLAMEGPU/FLAMEGPU2#857]
After #850 is merged, c++17 `std:any` can be used to replace `flamegpu::util::Any`. 

This is an API break due to use in prototypes of non-detail classes, but users should not directly be using our `Any` implementation, so low risk of causing issues but better to do sooner rather than later. 

This will conflict with #846, so is being deferred until after that work has completed.

This was formerly a sub-task of #611.

**Comment by Robadob** [comment]
As noted in Slack, challenge here is our internal `any` allows it to be initialised with purely a type index and data. Whilst this has much the same internal representation as `std::any`, the latter cannot be created without presence of a template argument.

As such, to use `std::any`, we would need to create a `std::any` to hold the result when the user first calls something in the public API, and pass this around/clone it when actually required later on.

Alternatively we could manually create some empty `std::any` which could be cloned for any common types, however that's a bit grim.

A further possible issue would be arrays, our internal version allows any to represent an array. I don't believe this is possible with `std::any` short of wrapping a `std::array` (or more likely `std::vector` as the length may not be available at compile time.
